### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by kingsleyzissou**
This is a quick prototype of how we could refactor the scheduler. It's in JS for now, just so we have an example of how this could be done.

### CODE REVIEW

**Comment by kingsleyzissou** [comment]
@wtrocki @Eunovo

**Comment by kingsleyzissou** [comment]
Added optimistic response capabilities.

There is a caveat. Deleting a todo item before the page has been refreshed sends the client id back to the server, which causes an error. So I need to fix that.

**Comment by kingsleyzissou** [comment]
@wtrocki this is more or less functional at this point. I guess the next thing would be to bring in idblocalstorage and test relationships.

There are a couple of todos too. We can look at this on monday or tuesday!

File: src/utils/Queue.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import { BehaviorSubject } from 'rxjs';
```
Comment (wtrocki) [inline src/utils/Queue.js:1]: rxjs :)

File: src/utils/Queue.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      filter(res => res.action === 'queued')
+    ).subscribe(observer);
+  }
+
+  /**
+   * Filter events for requeued events only
+   * 
+   * @param observer 
+   */
+  onRequeue(observer) {
```
Comment (wtrocki) [inline src/utils/Queue.js:107]: In what situation we will requeue? I though the whole purpose is to not queue stuff again

File: src/utils/Queue.js (hunk 1 above)
Comment (kingsleyzissou) [reply src/utils/Queue.js:1]: It’s amazing!

File: src/utils/Queue.js (hunk 2 above)
Comment (kingsleyzissou) [reply src/utils/Queue.js:107]: Ahh yeah just a bad function name. It’s not really requeueing, it’s for handling the reexecution when the app goes back online.

We can chat about this this week and go through things together

**Comment by wtrocki** [comment]
@kingsleyzissou I will use parts of this refactor and copy this code inside datastore. 
Thank you so much for amazing work!

**Comment by kingsleyzissou** [comment]
@wtrocki Yeah no problem. For the most part I used this code for the data store stuff. Just need to fix some of the typings and one or two smaller issues

**Comment by wtrocki** [comment]
I have been playing with scheduler and it seems that it has absolutely no value for the datastore replication. The way scheduler is designed would work mostly with the generic clients.

Main problems:

- Datastore needs to work with one single database that has both queue and original data
- DataStore operation with replication will affect both original objects and queue and we need transations - without them we can provide data integrity. 
- There is tons of legacy elements in the scheduler that do not fit to the.

The only file that I found useful for the original scheduler was IDBLocalstore but I have modified it heavily.

**Pull request closed without merging**